### PR TITLE
feat(backstage-plugin-argo-cd-backend): support application in any namespace.

### PR DESCRIPTION
**Pull request by mugioka**
Closes https://github.com/RoadieHQ/roadie-backstage-plugins/issues/1222.

#### :heavy_check_mark: Checklist

- [ ] Added tests for new functionality and regression tests for bug fixes
- [x] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [ ] Added or updated documentation (if applicable)

### CODE REVIEW

File: plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts [hunk 1]
```diff
@@ -83,15 +85,12 @@ export class ArgoService implements ArgoServiceApi {
   async getRevisionData(
```
Comment (mugioka) [inline plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts:85]: this api does not support selector.
image

File: plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts [hunk 2]
```diff
@@ -116,16 +117,25 @@ export class ArgoService implements ArgoServiceApi {
...
         let getArgoAppDataResp: any;
+        let token: string;
+        try {
+          token = argoInstance.token || (await this.getArgoToken(argoInstance));
+        } catch (error: any) {
+          this.logger.error(
+            `Error getting token from Argo Instance ${argoInstance.name}: ${error.message}`,
+          );
+          return null;
+        }
```
Comment (mugioka) [inline plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts:136]: I think try/catch for token acquisition/Application data acquisition should be separated in order to convey the error to the user in an understandable manner.

File: plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts [hunk 3]
```diff
@@ -83,15 +85,12 @@ export class ArgoService implements ArgoServiceApi {
   async getRevisionData(
     baseUrl: string,
     options: {
-      name?: string;
-      selector?: string;
```
Comment (Xantier) [inline plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts:87]: I guess this `selector` was never actually filled in on any calling site?

Comment (mugioka) [reply plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts:87]: yes.

**Review by Xantier — APPROVED** [review]

**Comment by mugioka** [comment]
@Xantier 
https://github.com/RoadieHQ/roadie-backstage-plugins/actions/runs/7540603375/job/20525553890
The above action seems to have failed.
Is there anything I should do?

**Pull request merged**